src/core/mac/sub_mac.cpp

```diff
@@ -394,13 +394,13 @@ void SubMac::StartCsmaBackoff(void)
 
         if (ShouldHandleTransmitTargetTime())
         {
-            if (Time(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance()))) <
-                Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) + mTransmitFrame.mInfo.mTxInfo.mTxDelay -
-                    kCcaSampleInterval - kCslTransmitTimeAhead - kRadioHeaderShrDuration)
+            uint32_t aheadTime      = kCcaSampleInterval + kCslTransmitTimeAhead + kRadioHeaderShrDuration;
+            Time     timerStartTime = Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) - aheadTime;
+            Time     txStartTime    = timerStartTime + mTransmitFrame.mInfo.mTxInfo.mTxDelay;
+
+            if (Time(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance()))) < txStartTime)
             {
-                mTimer.StartAt(Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) - kCcaSampleInterval -
-                                   kCslTransmitTimeAhead - kRadioHeaderShrDuration,
-                               mTransmitFrame.mInfo.mTxInfo.mTxDelay);
+                TimerStartAt(timerStartTime, mTransmitFrame.mInfo.mTxInfo.mTxDelay);
```

Comment:
We may be able to simplify this and use `Timer::FireAt()` instead with `txStartTime` directly?

The `Timer::StartAt()` would add the start time with delay to calculate the fire time, which we already know here.